camayoc/tests/qcs/api/credentials/test_creds_common.py

```diff
@@ -0,0 +1,257 @@
+    :expectedresults: A new credential entry is created with the data.
+    """
+    cred = Credential(
+        cred_type=src_type,
+        client=shared_client,
+        password=uuid4())
+    cred.create()
+    # add the credential to the list to destroy after the test is done
+    cleanup.append(cred)
+    assert_matches_server(cred)
+
+
+@pytest.mark.parametrize('src_type', QCS_SOURCE_TYPES)
+@pytest.mark.parametrize('field', ['username', 'password'])
+def test_update(src_type, field, shared_client, cleanup):
+    """Create a credential and then update its username.
+
+    :id: 73ed2ed5-e623-48ec-9ea6-153017464d9c
+    :description: Create a credential with password, then update its
+        username.
+    :steps:
+        1) Create a credential with a username and password.
+        2) Update the credential with a new username or password.
+        3) Confirm credential has been updated.
+    :expectedresults: The credential is updated.
+    """
+    cred = Credential(
+        cred_type=src_type,
+        client=shared_client,
+        password=uuid4())
+    cred.create()
+    # add the id to the list to destroy after the test is done
+    cleanup.append(cred)
+    assert_matches_server(cred)
+
+    # give the cred a new value for the field
+    setattr(cred, field, uuid4())
+    cred.update()
+    assert_matches_server(cred)
+
+
+@pytest.mark.skip
+@pytest.mark.parametrize('src_type', QCS_SOURCE_TYPES)
+def test_negative_update_to_invalid(
+    src_type,
+    shared_client,
+    cleanup,
+    isolated_filesystem
+):
+    """Attempt to update valid credential with invalid data.
+
+    :id: d3002d47-daee-4fcc-ac7c-477738ffc447
+    :description: Create valid credentials, then attempt to update to be
+        invalid.
+    :steps:
+        1) Create valid credentials with passwords.
+        2) Update the credentials:
+            a) missing usernames
+            c) missing password
+    :expectedresults: Error codes are returned and the credentials are
+        not updated.
+    :caseautomation: notautomated
+    """
+    pass
+
+
+@pytest.mark.parametrize('src_type', QCS_SOURCE_TYPES)
+def test_negative_create_no_name(src_type, cleanup):
+    """Attempt to create a credential missing a name.
+
+    The request should be met with a 4XX response.
+
+    :id: faf2d9fd-8b19-4bf7-b4a9-761da6de34e4
+    :description: Create a credential missing a name.
+    :steps: Send POST with necessary data to documented api endpoint.
+    :expectedresults: Error is thrown and no new credential is created.
+    """
+    client = api.Client(api.echo_handler)
+    cred = Credential(
+        cred_type=src_type,
+        client=client,
+        username='',
+        password=uuid4(),
+    )
+    response = cred.create()
+    assert response.status_code == 400
+    assert cred._id is None
+
+
+@pytest.mark.parametrize('src_type', QCS_SOURCE_TYPES)
+def test_negative_create_missing(src_type, cleanup):
```

Comment:
To follow the name of the above test, what about `test_negative_create_no_password`?

Reply:
Also both tests are pretty similar, maybe you can consider using a parameter for the field to be missing.